src/components/MultiFilterButton/index.tsx

```diff
@@ -0,0 +1,81 @@
+import { Link } from 'react-router-dom'
+import { Popover } from 'antd'
+import { useTranslation } from 'react-i18next'
+import { ReactComponent as FilterIcon } from '../../assets/filter_icon.svg'
+import { ReactComponent as SelectedIcon } from '../../assets/selected-icon.svg'
+import { ReactComponent as NotSelectedIcon } from '../../assets/not-selected-icon.svg'
+import { ReactComponent as PartialSelectedIcon } from '../../assets/partial-selected-icon.svg'
+import { useSearchParams } from '../../hooks'
+import styles from './styles.module.scss'
+
+export function MultiFilterButton({
+  filteredList,
+  isMobile,
+  filterName,
+}: {
+  filterName: string
+  filteredList: { key: string; value: string; to: string; title: string | JSX.Element }[]
+  isMobile?: boolean
+}) {
+  const { t } = useTranslation()
+  const params = useSearchParams(filterName)
+  const filter = params[filterName]
+  let types = filteredList.map(f => f.value)
+  if (filter !== undefined) {
+    types = filter.split(',').filter(t => t !== '')
+  }
+
+  return (
+    <Popover
+      className={styles.container}
+      placement="bottomRight"
+      trigger={isMobile ? 'click' : 'hover'}
+      overlayClassName={styles.antPopover}
+      content={
+        <div className={styles.filterItems}>
+          <div className={styles.selectTitle}>
+            <h2>{t('components.multi_filter_button.select')}</h2>
+            <Link
+              key="all"
+              to={() => {
+                return types.length === 0 || types.length !== filteredList.length
+                  ? filteredList[0].to
+                  : `${filteredList[0].to}?${new URLSearchParams({ [filterName]: '' }).toString()}`
+              }}
+            >
+              {types.length > 0 ? (
+                <>{types.length === filteredList.length ? <SelectedIcon /> : <PartialSelectedIcon />}</>
```

Comment:
Unnecessary `<>` wrap

Reply:
`types.length === filteredList.length` appears in 2 places, and it can be named `isAllSelected` to make the code clear.